--- lib/error.cc	2021-01-12 15:45:08.102361256 +1100
+++ lib/error.cc	2021-01-12 15:48:19.231653870 +1100
@@ -32,12 +32,21 @@
 #if CLICK_USERLEVEL || CLICK_TOOL || CLICK_MINIOS
 # include <unistd.h>
 #endif
+
+#ifdef HAVE_VA_LIST_AS_ARRAY
+#define MAKE_POINTER_FROM_VA_LIST_ARG(arg) ((va_list *)(arg))
+#else
+#define MAKE_POINTER_FROM_VA_LIST_ARG(arg) (&(arg))
+#endif
+
 CLICK_DECLS
 
 /** @file error.hh
  * @brief Flexible error handling classes.
  */
 
+
+
 struct ErrorHandler::Conversion {
     String name;
     ConversionFunction hook;
@@ -728,7 +737,11 @@
 	    s = s2 + 1;
 	    for (Conversion *item = error_items; item; item = item->next)
 		if (item->name.equals(s1, s2 - s1)) {
+#ifdef HAVE_VA_LIST_AS_ARRAY
+		    strstore = item->hook(flags, (va_list*) VA_LIST_REF(val));
+#else
 		    strstore = item->hook(flags, VA_LIST_REF(val));
+#endif
 		    s1 = strstore.begin();
 		    s2 = strstore.end();
 		    goto got_result;
